templates/test/ci/prow/kustomization.yaml

```diff
@@ -18,6 +18,7 @@ patchesStrategicMerge:
   - ../patches/machine-deployment-worker-counts.yaml
   - ../../../azure-cluster-identity/azurecluster-identity-ref.yaml
   - ../../../flavors/base-windows-containerd/kubeadm-control-plane.yaml
+  - ../patches/azuremachinetemplate-vmextension.yaml
```

Comment:
so I'm slightly concerned about the number of templates this change ended up modifying, given that the "prow" test flavor is used as the basis for most other things... another argument for restructuring our test templates 

Not sure how we can avoid this without doing a major refactor of templates though, and there isn't a single template that makes more sense to use as _the_  test template for extensions.

Reply:
That's true. I considered creating a whole new template just for testing extensions, but that might be too much make the E2E test run slower. Adding the test to the default template made the most sense to me, but it may be better design-wise to incorporate it into a different test.

Reply:
Let's not tackle this in this PR but we're overdue for a template cleanup/simplification. Some of the cluster test templates could probably be combined (anything that tests a machine / machinepool feature doesn't need its own cluster)